tests/test_router_bgp.rb

```diff
@@ -540,31 +538,32 @@ def test_routerbgp_set_get_confederation_peers
       # confederation id be configured first so the expectation
       # in the next test is an empty peer list
       bgp.confederation_id = 55
+
       assert_empty(bgp.confederation_peers,
                    "vrf #{vrf}: bgp confederation_peers list should be empty")
-      bgp.confederation_peers_set(15)
+      bgp.confederation_peers = (15)
       assert_equal('15', bgp.confederation_peers,
                    "vrf #{vrf}: bgp confederation_peers list should be '15'")
-      bgp.confederation_peers_set(16)
+      bgp.confederation_peers = (16)
       assert_equal('16', bgp.confederation_peers,
                    "vrf #{vrf}: bgp confederation_peers list should be '16'")
-      bgp.confederation_peers_set(55.77)
+      bgp.confederation_peers = (55.77)
       assert_equal('55.77', bgp.confederation_peers,
-                   "vrf #{vrf}: bgp confederation_peers list should be" \
+                   "vrf #{vrf}: bgp confederation_peers list should be " \
                    "'55.77'")
-      bgp.confederation_peers_set('15 16 55.77 18 555 299')
-      assert_equal('15 16 55.77 18 555 299',
+      bgp.confederation_peers = ('15 16 55.77 18 555 299')
```

Comment:
But what about: bgp.confederation_peers_set(15) should this be: bgp.confederation_peers_set([15])?

And if we are moving to setting with an array - doesn't that affect the manifest? So instead of:

confed_peers => '1 2 3'

now:

confed_peers => "['1', '2', '3']"

Sort of the whole point was: input=string, convert to array so it can be sorted etc, output=string